hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeBenchMark.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.internal.util.phonehome;
+
+import com.hazelcast.config.AttributeConfig;
+import com.hazelcast.config.CacheConfig;
+import com.hazelcast.config.EvictionConfig;
+import com.hazelcast.config.EvictionPolicy;
+import com.hazelcast.config.HotRestartConfig;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.QueryCacheConfig;
+import com.hazelcast.config.WanReplicationRef;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.map.IMap;
+import com.hazelcast.test.HazelcastTestSupport;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+
+import javax.cache.CacheManager;
+import javax.cache.spi.CachingProvider;
+
+import static com.hazelcast.cache.CacheTestSupport.createServerCachingProvider;
+import static com.hazelcast.test.Accessors.getNode;
+
+@State(Scope.Benchmark)
+public class PhoneHomeBenchMark extends HazelcastTestSupport {
+
+    private PhoneHome phoneHome;
+    private Node node;
+
+    @Setup
+    public void setup() {
+        HazelcastInstance hz = createHazelcastInstance();
+        node = getNode(hz);
+        phoneHome = new PhoneHome(node);
+        CachingProvider cachingProvider = createServerCachingProvider(hz);
+        CacheManager cacheManager = cachingProvider.getCacheManager();
+
+        for (int i = 1; i <= 10000; i++) {
+            hz.getMap("maps" + i);
+            hz.getSet("set" + i);
+            hz.getQueue("queue" + i);
+            hz.getMultiMap("multimap" + i);
+            hz.getList("list" + i);
+            hz.getRingbuffer("ringbuffer" + i);
+            hz.getTopic("topic" + i);
+            hz.getReplicatedMap("replicatedmap" + i);
+            hz.getCardinalityEstimator("cardinalityEstimator" + i);
+            hz.getPNCounter("PNcounter" + i);
+            hz.getFlakeIdGenerator("flakeid" + i);
+            cacheManager.createCache("cache" + i, new CacheConfig<>("cache" + i));
+        }
+
+        for (int i = 1; i <= 100000; i++) {
+            IMap<Object, Object> iMap = hz.getMap("map" + i);
+            MapConfig config = node.getConfig().getMapConfig("map" + i);
+
+            if (i % 10 == 0) {
+                config.getMapStoreConfig().setClassName(getClass().getName()).setEnabled(true);
```

Comment:
I doubt if `getClass().getName()` will work here. Does this class implement the `MapStore` interface.